security/Member.php

```diff
@@ -1842,31 +1860,35 @@ public function __construct() {
 	 * @return bool Returns TRUE if the submitted data is valid, otherwise
 	 *              FALSE.
 	 */
-	public function php($data) {
+	public function php($data)
+	{
 		$valid = parent::php($data);
 
-		$identifierField = Member::config()->unique_identifier_field;
-		$member = DataObject::get_one('Member', array(
-			"\"$identifierField\"" => $data[$identifierField]
-		));
+		$identifierField = (string)Member::config()->unique_identifier_field;
 
-		// if we are in a complex table field popup, use ctf[childID], else use ID
-		if(isset($_REQUEST['ctf']['childID'])) {
-			$id = $_REQUEST['ctf']['childID'];
-		} elseif(isset($_REQUEST['ID'])) {
-			$id = $_REQUEST['ID'];
-		} else {
-			$id = null;
+		$id = isset($data['ID']) ? (int)$data['ID'] : 0;
+		if(!$id && ($ctrl = $this->form->getController())){
+			// get the record when within GridField (Member editing page in CMS)
+			if($ctrl instanceof GridFieldDetailForm_ItemRequest && $record = $ctrl->getRecord()){
+				$id = $record->ID;
+				// set the ID to the data array, so that extensions can also use it
+				$data['ID'] = $id;
+			}
+		}
+
+		$members = Member::get()->filter($identifierField, $data[$identifierField]);
```

Comment:
No. You can just amend this PR and force push. The updates will come into the PR automatically